src/xrpld/app/tx/detail/MPTokenAuthorize.h

```diff
@@ -0,0 +1,63 @@
+//------------------------------------------------------------------------------
+/*
+  This file is part of rippled: https://github.com/ripple/rippled
+  Copyright (c) 2023 Ripple Labs Inc.
+
+  Permission to use, copy, modify, and/or distribute this software for any
+  purpose  with  or without fee is hereby granted, provided that the above
+  copyright notice and this permission notice appear in all copies.
+
+  THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+  WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+  MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+  ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+  WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+  ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+  OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_TX_MPTOKENAUTHORIZE_H_INCLUDED
+#define RIPPLE_TX_MPTOKENAUTHORIZE_H_INCLUDED
+
+#include <xrpld/app/tx/detail/Transactor.h>
+
+namespace ripple {
+
+struct MPTAuthorizeArgs
+{
+    XRPAmount const& priorBalance;
+    uint192 const& mptIssuanceID;
+    AccountID const& account;
+    std::uint32_t flags;
+    std::optional<AccountID> holderID;
+};
```

Comment:
This struct is not used anywhere outside of the `MPTokenAuthorize` class, so there's no need to expose it publicly. You can make it a `private` subclass of `MPTokenAuthorize`.

Reply:
I disagree. This is in the family of helpers just like `MPTokenIssuanceCreate::create`. There may be only one caller right now, but there may be more in the future. My `vault` branch uses `MPTokenIssuanceCreate::create`, for example.

There could be an argument for putting these helpers in `src/xrpld/ledger/View.h`, but I don't have a preference.